examples/image_captioning/predict.py

```diff
@@ -26,12 +28,27 @@ def main():
     parser.add_argument('--rnn', type=str, default='nsteplstm',
                         choices=['nsteplstm', 'lstm'],
                         help='Language model layer type')
-    parser.add_argument('--gpu', type=int, default=0,
-                        help='GPU ID (negative value indicates CPU)')
+    parser.add_argument('--device', '-d', type=str, default='-1',
+                        help='Device specifier. Either ChainerX device '
+                        'specifier or an integer. If non-negative integer, '
+                        'CuPy arrays with specified device id are used. If '
+                        'negative integer, NumPy arrays are used')
     parser.add_argument('--max-caption-length', type=int, default=30,
                         help='Maximum caption length generated')
+    group = parser.add_argument_group('deprecated arguments')
+    group.add_argument('--gpu', '-g', dest='device',
+                       type=int, nargs='?', const=0,
+                       help='GPU ID (negative value indicates CPU)')
     args = parser.parse_args()
 
+    device = chainer.get_device(args.device)
+    if device.xp is chainerx:
+        sys.stderr.write('This example does not support ChainerX devices.\n')
+        sys.exit(1)
```

Comment:
Is it the case that it actually works for ChainerX if `--rnn nsteplstm` ? I image that's why you also added `atleast_1d`. How about only existing for `--rnn lstm`?

Reply:
I don't think it's so productive to fine-tune the error to look for the cases in which ChainerX works.

Reply:
I think that is fine too but we should in that case not introduce `_atleast_1d` as it's never called.

Reply:
That's true. Removed `_atleast_1d`.